packages/angular/cli/BUILD.bazel

```diff
@@ -147,6 +147,7 @@ ts_library(
         "//packages/angular_devkit/schematics",
```

Comment:
Drive by comment. I have not actually checked the code.
But have we considered allowing an array of strings as an input and normalizing the options in the builder?

IE: 
In https://github.com/angular/angular-cli/blob/d87822c1cef972026e7ea52be7728e7076478911/packages/angular/build/src/builders/application/schema.json#L223
```
    "define": {
      "description": "Defines global identifiers that will be replaced with a specified constant value when found in any JavaScript or TypeScript code including libraries. The value will be used directly. String values must be put in quotes. Identifiers within Angular metadata such as Component Decorators will not be replaced.",
      "oneOf": [
        {
          "description": "Defines global identifiers",
          "type": "array",
          "items": {
            "type": "string",
            "pattern": "^.+=.+$"
          },
          "default": []
        },
        {
          "type": "object",
          "additionalProperties": {
            "type": "string"
          }
        }
      ]
    },
```

And in https://github.com/angular/angular-cli/blob/d87822c1cef972026e7ea52be7728e7076478911/packages/angular/build/src/builders/application/options.ts#L218

```ts
  let define: any;
  if (Array.isArray(options.define)) {
    define = {};
    for (const keyValue of options.define) {
      const [key, value] = keyValue.split('=')
      define[key] = value
    }
  } else {
    define = options.define;
  }

```

Reply:
I assume the question there is if we would want the alternate syntax in the config file as well? I tend to be hesitant when it comes to creating that kind of branching value space since it can make both documentation and automation harder down the line. Case in point: It may introduce `.split` bugs if some other code tries to interpret the key/value pairs in the array and doesn't consider string values that contain `=` inside of them.